### PR TITLE
EP-87560 - Bump activesupport to 6.1.7.3 & configure travis

### PR DESCRIPTION
**Pull request by abhishek-jadav98**
This PR includes
- Upgrading ruby version from 2.3 to 2.7.7
- Upgrading activesupport from 5.2.0 to 6.1.7.3
- Fixed `NameError: uninitialized constant CloudStorageInterface::AwsS3Interface::Aws` by requiring `aws-sdk-s3` to aws_s3_interface.rb. (Test cases were failing before too. checked `csi-0.5.5` branch)
- Fixed `/gcp_gcs_interface.rb:91: warning: URI.unescape is obsolete` warning by using `decode_www_form_component` (This is after ruby upgrade as unescape is deprecated in ruby 2.7.7)
-  Fixed failing test cases after ruby upgrade
- configuring travis for test suite execution

[JIRA](https://edcast.atlassian.net/browse/EP-87650)

### CODE REVIEW

File: lib/cloud_storage_interface/gcp_gcs_interface.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ def presigned_post(bucket_name:, key:, acl:, success_action_status:, expiration:
       # Wierd inconsistency between S3 and GCS APIs.
       # GCS escapes ${filename} in the key before returning it in the fields.
       # We need to manually unescape it.
-      post_obj.fields[:key] = URI.unescape(post_obj.fields[:key])
+      post_obj.fields[:key] = URI.decode_www_form_component(post_obj.fields[:key])
```
Comment (abhishek-jadav98) [inline lib/cloud_storage_interface/gcp_gcs_interface.rb:91]: Reference - https://ruby-doc.org/stdlib-2.5.3/libdoc/uri/rdoc/URI/Escape.html#method-i-unescape

**Comment by abhishek-jadav98** [comment]
Note - Process followed to raise this PR

- Updated activesupport and ruby version.
- Updated Version file from 0.0.31 to 0.0.32
- ran bundle install
- made code changes (i.e fixed failing test case & requiring aws sdk)
- ran `gem build cloud_storage_interface.gemspec`
- deleted `cloud_storage_interface-0.0.31.gem` and commited `cloud_storage_interface-0.0.32.gem` file
- commited and pushed here

**Review by nitinsingh25 — APPROVED** [review]

File: lib/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module CloudStorageInterface
-  VERSION = '0.0.31'
+  VERSION = '0.0.32'
```
Comment (nitinsingh25) [inline lib/version.rb:2]: @abhishek-jadav98 Change its version to 1.0.0, since it is major upgrade.

Comment (abhishek-jadav98) [reply lib/version.rb]: @nitinsingh25, version upgraded. Please review.

**Review by nitinsingh25 — APPROVED** [review]

**Pull request merged**